src/FFNNWaveFunction.cpp

```diff
@@ -0,0 +1,89 @@
+#include "FFNNWaveFunction.hpp"
+
+#include <cmath>
+
+
+
+// --- interface for manipulating the variational parameters
+void FFNNWaveFunction::setVP(const double *vp){
+    for (int i=0; i<_ffnn->getNBeta(); ++i)
+        _ffnn->setBeta(i, vp[i]);
+}
+
+
+void FFNNWaveFunction::getVP(double *vp){
+    for (int i=0; i<_ffnn->getNBeta(); ++i)
+        vp[i] = _ffnn->getBeta(i);
+}
+
+
+
+
+// --- methods herited from MCISamplingFunctionInterface
+
+void FFNNWaveFunction::samplingFunction(const double * in, double * out){
+    _ffnn->setInput(_ffnn->getNInput(), in);
+    _ffnn->FFPropagate();
+    out[0] = pow(_ffnn->getOutput(1), 2);
+}
+
+
+double FFNNWaveFunction::getAcceptance(){
+    if ((getProtoOld(0) == 0.) && (getProtoNew(0) != 0.)){
+       return 1.;
+    } else if ((getProtoOld(0) == 0.) && (getProtoNew(0) == 0.)) {
+       return 0.;
+    }
+
+    return getProtoNew(0)/getProtoOld(0);
+}
+
+
+
+
+// --- wf derivatives
+
+double FFNNWaveFunction::d1(const int &i, const double *in){
+    _ffnn->setInput(_ffnn->getNInput(), in);
+    _ffnn->FFPropagate();
+    return _ffnn->getFirstDerivative(1, i)/_ffnn->getOutput(1);
+}
+
+
+double FFNNWaveFunction::d2(const int &i, const int &j, const double *in){
+    if (i != j)
```

Comment:
I think we should change the abstract parent class and remove the j, and allow only pure second derivs. As you said: When are we going to need it?
Of course maybe we still want to keep the possibility to use them in the future...

Reply:
we should remove it yes, but that's another Trello story (already on the board btw)